### ISSUE TITLE
java.lang.NoSuchMethodError: com.google.common.cache.CacheBuilder.expireAfterWrite(Ljava/time/Duration;)

### ISSUE THREAD

**Issue by auroranil** [GoogleCloudDataproc/hadoop-connectors#383]
Caller:
https://github.com/GoogleCloudDataproc/hadoop-connectors/blob/master/gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageImpl.java#L142

```
CacheBuilder.newBuilder()
          .expireAfterWrite(Duration.ofHours(1))
```

Callee:
https://github.com/google/guava/blob/master/android/guava/src/com/google/common/cache/CacheBuilder.java#L654
```
public CacheBuilder<K, V> expireAfterWrite(long duration, TimeUnit unit) {
    checkState(
        expireAfterWriteNanos == UNSET_INT,
        "expireAfterWrite was already set to %s ns",
        expireAfterWriteNanos);
    checkArgument(duration >= 0, "duration cannot be negative: %s %s", duration, unit);
    this.expireAfterWriteNanos = unit.toNanos(duration);
    return this;
  }
```

Relevant Stackoverflow link which explains the issue: https://stackoverflow.com/questions/61810399/execute-google-cloud-dataflow-pipeline-from-spring-controller

To summarise, the method requires an additional parameter `TimeUnit unit`, which is not specified in the caller. I could not find any overloaded method which just has one parameter.

**Comment by auroranil** [comment]
Here is where it was changed: https://github.com/GoogleCloudDataproc/hadoop-connectors/commit/29528a9784e77e76b30bffa5272354a73b20f3dd#diff-272b8441afbd98ca400ec66e99c17fb8

**Comment by medb** [comment]
In what environment are you running your application? And what library do you use to access GCS, [GCSIO](https://search.maven.org/artifact/com.google.cloud.bigdataoss/gcsio) or [GCS connector](https://search.maven.org/artifact/com.google.cloud.bigdataoss/gcs-connector)?

**Comment by medb** [comment]
Posted an answer on StackOverflow: https://stackoverflow.com/a/61825568/3227693

**Comment by medb** [comment]
This is a duplicate of https://github.com/GoogleCloudDataproc/hadoop-connectors/issues/307 request.

**Comment by auroranil** [comment]
It is a Spring Boot application with dependencies specified in `build.gradle` file. GCSIO is being used by Apache Beam. The versions I am using are specified here below:

```
dependencies {
    compile group: 'org.apache.beam', name: 'beam-sdks-java-core', version: '2.20.0'
    compile group: 'org.apache.beam', name: 'beam-runners-google-cloud-dataflow-java', version: '2.20.0'
}
```

I was able to resolve this issue by adding guava version `29-0-jre` to `build.gradle` file:

```
dependencies {
    compile group: 'org.apache.beam', name: 'beam-sdks-java-core', version: '2.20.0'
    compile group: 'org.apache.beam', name: 'beam-runners-google-cloud-dataflow-java', version: '2.20.0'
    compile group: 'com.google.guava', name: 'guava', version: '29.0-jre'
}
```

**Comment by mortiz143** [comment]
fixed it with this,
```
configurations.all {
    resolutionStrategy {
        force 'com.google.guava:guava:30.1.1-jre'
    }
}
```

**Comment by Khushboo1192** [comment]
@mortiz143 I recently updated from 2.18 to 2.29 version for apache beam and now I am getting the above error, I tried forcing jre version as per your recommendation but I cannot resolve the error. Could you help why is that happening? Thanks!

**Comment by fr-amitverma** [comment]
for built.sbt file, following works

dependencyOverrides += "com.google.guava" % "guava" % "30.1-jre"
